pkg/webhook/workspace/handler/workspace.go

```diff
@@ -45,6 +45,10 @@ func (h *WebhookHandler) MutateWorkspaceOnUpdate(_ context.Context, req admissio
 	if err != nil {
 		return admission.Errored(http.StatusBadRequest, err)
 	}
+	immutable := oldWksp.Annotations[config.WorkspaceImmutableAnnotation]
+	if immutable == "true" {
+		return admission.Denied(fmt.Sprintf("workspace '%s' is immutable. To make modifications it must be deleted and recreated", oldWksp.Name))
```

Comment:
Probably we need to allow changing `spec.started` attributed even for immutable workspaces.
It's not really actual for cloud-shell workspace but do you think in future it makes sense to deny changing commands as well?

Reply:
Yeah I imagine once the started field is implemented this can be revisited. This PR is more of a question: is it worth our while adding this and supporting it? I can add an 'limited-mutation' branch to the webhook but don't want to spend the time on it if we don't want the overall feature.